packages/core/src/shared/telemetry/spans.ts

```diff
@@ -90,6 +90,40 @@ function getValidatedState(state: Partial<MetricBase>, definition: MetricDefinit
     return missingFields.length !== 0 ? Object.assign({ missingFields }, state) : state
 }
 
+/**
+ * Options used for the creation of a span
+ */
+export type SpanOptions = {
+    /** True if this span should emit its telemetry events. Defaults to true if undefined. */
+    emit?: boolean
+
+    /**
+     * Adds a function entry to the span stack.
+     *
+     * This allows you to eventually retrieve the function entry stack by using {@link TelemetryTracer.getFunctionStack()},
+     * which tells you the chain of function executions to bring you to that point in the code.
+     *
+     * Example:
+     * ```
+     * function a() {
+     *   telemetry.your_Metric.run(() => b(), { function: { name: 'a'} })
+     * }
+     *
+     * function b() {
+     *   telemetry.your_Metric.run(() => c(), { function: { name: 'b'} })
+     * }
+     *
+     * function c() {
+     *   telemetry.your_Metric.run(() => {
+     *     const stack = telemetry.getFunctionStack()
+     *     console.log(stack) // [ {source: 'a' }, { source: 'b' }, { source: 'c' }]
+     *   }, { function: { name: 'c'} })
+     * }
+     * ```
+     */
+    function?: FunctionEntry
```

Comment:
I wonder if this could be functionId? So a function ids is a a single entry in a stack of function ids? naming is hard 🙃

Reply:
done